packages/ng-table-builder/src/table-builder.component.ts

```diff
@@ -115,6 +119,34 @@ export class TableBuilderComponent<T>
         return this.sourceRef.length > 0;
     }
 
+    public get rootHeight(): string {
+        const height: Nullable<string | number> = this.expandableTableExpanded
+            ? this.height
+            : this.headerClientHeight + this.footerClientHeight;
+        if (checkValueIsFilled(height)) {
+            const heightAsNumber: number = Number(height);
+            return isNaN(heightAsNumber) ? String(height) : `${height}px`;
+        } else {
+            return '';
+        }
+    }
+
+    public get headerClientHeight(): number {
+        return this.headerRef?.nativeElement?.clientHeight || 0;
+    }
+
+    public get footerClientHeight(): number {
+        return this.footerRef?.nativeElement?.clientHeight || 0;
+    }
+
+    private get expandableTableExpanded(): boolean {
```

Comment:
can we use pipe instead getter?

Reply:
I did not use this getter in the template.

Reply:
oh, sorry